lib/scheduler.pm

```diff
@@ -122,10 +121,15 @@ sub load_yaml_schedule {
     if (my $yamlfile = get_var('YAML_SCHEDULE')) {
         my $schedule      = $ypp->load_file($root_project_dir . $yamlfile);
         my %schedule_vars = parse_vars($schedule);
+        my $test_context_instance = undef;
         while (my ($var, $value) = each %schedule_vars) { set_var($var, $value) }
         my @schedule_modules = parse_schedule($schedule);
         parse_test_suite_data($schedule);
-        loadtest($_) for (@schedule_modules);
+        if ($schedule_vars{TEST_CONTEXT}) {
```

Comment:
Why not to allow setting this variable outside of the schedule, meaning why not to use get_var?

Reply:
the structure is already presented in the stack of `load_yaml_schedule` by `parse_vars. Which pretty much does exactly that for us.
```
 $value =~ s/%(.*?)%/get_var($1)/eg
 $vars{$var} = $value;
```.
Make another call to `get_var` seems unnecessarily.

Reply:
Please, read my comment again, it's unnecessary if you allow setting this variable only in yaml files. In case `TEST_CONTEXT` is set e.g. in the job group, you won't pass the context. I don't think that we should have this constraint.

Reply:
fixed. i am checking first if it is in the hash otherwise it checks the variable